macOSUpgrade.sh

```diff
@@ -69,15 +69,21 @@
 ## Specify path to OS installer – Use Parameter 4 in the JSS, or specify here.
 ## Parameter Label: Path to the macOS installer
 ## Example: /Applications/Install macOS High Sierra.app
-OSInstaller="$4"
+OSInstaller="$( echo "$4" | /usr/bin/xargs )"
 
 ## Version of Installer OS. Use Parameter 5 in the JSS, or specify here.
 ## Parameter Label: Version of macOS
 ## Example Command: /usr/libexec/PlistBuddy -c 'Print :"System Image Info":version' "/Applications/Install macOS High Sierra.app/Contents/SharedSupport/InstallInfo.plist"
 ## Example: 10.12.5
 installerVersion="$5"
-installerVersionMajor=$( /bin/echo "$installerVersion" | /usr/bin/awk -F. '{print $2}' )
-installerVersionMinor=$( /bin/echo "$installerVersion" | /usr/bin/awk -F. '{print $3}' )
+installerVersionNumber="$( /bin/echo "$installerVersion" | /usr/bin/awk -F. 'N = $1 * 10 ** 4 + $2 * 10 ** 2 + $3 {print N}' )"
```

Comment:
Since the logic of the verdict is getting more complex, why not change the runtime to zsh and use `is-at-least` function as a build-in?

One of the factors is that the default shell is now zsh from Catalina.

Reply:
Wouldn't it be a good time to change the shell from bash to zsh, stop supporting older OS versions, and clean up the code of this script?

Reply:
I do agree and it's something I've wanted to do, just haven't had the time. 

I'll review this yet today and get it merged pending no issues. But I would love to work on moving this to zsh to ensure compatibility going forward. If you have the time, feel free to start. As always, your work on this project is greatly appreciated!